src/mscorlib/src/System/Globalization/EncodingTable.Unix.cs

```diff
@@ -1,101 +1,181 @@
+            { "ISO-10646-UCS-2", 1200 },
+            { "iso-8859-1", 28591 },
+            { "iso-ir-100", 28591 },
+            { "iso-ir-6", 20127 },
+            { "ISO646-US", 20127 },
+            { "iso8859-1", 28591 },
+            { "ISO_646.irv:1991", 20127 },
+            { "iso_8859-1", 28591 },
+            { "iso_8859-1:1987", 28591 },
+            { "l1", 28591 },
+            { "latin1", 28591 },
+            { "ucs-2", 1200 },
+            { "unicode", 1200},
+            { "unicode-1-1-utf-7", 65000 },
+            { "unicode-1-1-utf-8", 65001 },
+            { "unicode-2-0-utf-7", 65000 },
+            { "unicode-2-0-utf-8", 65001 },
+            // People get confused about the FFFE here.  We can't change this because it'd break existing apps.
+            // This has been this way for a long time, including in Mlang.
+            // Big Endian, BOM seems backwards, think of the BOM in little endian order.
+            { "unicodeFFFE", 1201},
+            { "us", 20127 },
+            { "us-ascii", 20127 },
+            { "utf-16", 1200 },
+            { "UTF-16BE", 1201},
+            { "UTF-16LE", 1200},
+            { "utf-32", 12000 },
+            { "UTF-32BE", 12001 },
+            { "UTF-32LE", 12000 },
+            { "utf-7", 65000 },
+            { "utf-8", 65001 },
+            { "x-unicode-1-1-utf-7", 65000 },
+            { "x-unicode-1-1-utf-8", 65001 },
+            { "x-unicode-2-0-utf-7", 65000 },
+            { "x-unicode-2-0-utf-8", 65001 },
+        };
 
-                default:
-                    return new CodePageDataItem("utf-8", CodePageUtf8, "utf-8", "utf-8", 771);
+        private struct InternalCodePageDataItem
+        {
+            public InternalCodePageDataItem(ushort codePage, ushort uiFamilyCodePage, string name, uint flags)
+            {
+                CodePage = codePage;
+                UiFamilyCodePage = uiFamilyCodePage;
+                Name = name;
+                Flags = flags;
             }
+
+            public ushort CodePage { get; private set; }
+            public ushort UiFamilyCodePage { get; private set; }
+            public string Name { get; private set; }
+            public uint Flags { get; private set; }
         }
 
-        // PAL ends here.
+        // redeclaring these constants here for readability below
+        private const uint MIMECONTF_MAILNEWS = Encoding.MIMECONTF_MAILNEWS;
+        private const uint MIMECONTF_BROWSER = Encoding.MIMECONTF_BROWSER;
+        private const uint MIMECONTF_SAVABLE_MAILNEWS = Encoding.MIMECONTF_SAVABLE_MAILNEWS;
+        private const uint MIMECONTF_SAVABLE_BROWSER = Encoding.MIMECONTF_SAVABLE_BROWSER;
 
-        const int CodePageUtf7 = 65000;
-        const int CodePageUtf8 = 65001;
-        const int CodePageUtf16 = 1200;
-        const int CodePageUtf16BE = 1201;
-        const int CodePageUtf32 = 12000;
+        private static InternalCodePageDataItem[] encodingDataTableItems = new[]
```

Comment:
One thing I just realized here is that we can just cache static instances of CodePageDataItem, and get rid of the InternalCodePageDataItem altogether.  CodePageDataItem is internal and immutable, so there shouldn't be any problem handing out our cached instances.